lib/matplotlib/axes/_axes.py

```diff
@@ -5287,12 +5275,27 @@ def imshow(self, X, cmap=None, norm=None, aspect=None,
             The Colormap instance or registered colormap name used to map
             scalar data to colors. This parameter is ignored for RGB(A) data.
 
-        norm : `~matplotlib.colors.Normalize`, optional
+        norm : str or `~matplotlib.colors.Normalize`, optional
             The `.Normalize` instance used to scale scalar data to the [0, 1]
             range before mapping to colors using *cmap*. By default, a linear
             scaling mapping the lowest value to 0 and the highest to 1 is used.
             This parameter is ignored for RGB(A) data.
 
+            The normalization can also be given as a str, which should be a
+            scale name (as in `~.Axes.set_xscale`, i.e. one of "linear", "log",
+            "symlog", "logit", etc.). In that case, a normalization class is
+            dynamically generated from the corresponding scale, as if using
+            that scale for the image's colorbar.
```

Comment:
```suggestion
            If given, this can be
            
            - a `.Normalize` instance, or subclass thereof.
              See :doc:`/tutorials/colors/colormapnorms`.
            - a scale name, i.e. one of "linear", "log", "symlog", "logit"
              etc. For a full list of available scales call
              `matplotlib.scales.get_scale_names()`.              
              In that case, a suitable `.Normalize` subclass is dynamically
              generated and an instance thereof is returned.
```

Reasoning:
- Structure/readability: The two bullet points reflect the to possible types.
- IMHO it does not help to reference `set_xscale`. That doesn't contain helpful additional information. If we want have a scale reference, that should be linked. If not `get_scale_names()` is the best we can do.
- I think we don't have to explicitly mention that the corresponding scale is used for the colorbar. That's the intuitively expected behavior. Everything else would we weird. Leave out --> less clutter. But also ok, if you want to explicit mention it.